python-frontend/src/main/java/org/sonar/plugins/python/api/symbols/Symbol.java

```diff
@@ -35,6 +36,10 @@
 
   Kind kind();
 
+  @Beta
+  @CheckForNull
+  Symbol annotatedType();
```

Comment:
I think we should be consistent between `returnTypeAnnotation()` and `annotatedType()`: 
- either we return `Optional<Symbol>` for both 
- or we return a nullable Symbol for both.

Nitpick: Also names don't sound consistent: 
- either 'returnTypeAnnotation()' and 'typeAnnotation()' 
- or 'annotatedReturnType()' and 'annotatedType()'